src/Juvix/Compiler/Backend/Geb/Evaluator/Data/Values.hs

```diff
@@ -0,0 +1,103 @@
+module Juvix.Compiler.Backend.Geb.Evaluator.Data.Values where
+
+import Control.DeepSeq
+import GHC.Show
+import Juvix.Compiler.Backend.Geb.Data.Context as Context
+import Juvix.Compiler.Backend.Geb.Language hiding (show)
+
+data GebValue
+  = GebValueMorphismUnit
+  | GebValueMorphismInteger Integer
+  | GebValueMorphismLeft GebValue
+  | GebValueMorphismRight GebValue
+  | GebValueMorphismPair ValueMorphismPair
+  | GebValueClosure ValueClosure
+  deriving stock (Eq, Generic)
+
+instance NFData GebValue
+
+data ValueMorphismPair = ValueMorphismPair
+  { _valueMorphismPairLeft :: GebValue,
+    _valueMorphismPairRight :: GebValue
+  }
+  deriving stock (Eq, Generic)
+
+instance NFData ValueMorphismPair
+
+data ValueMorphismCase = ValueMorphismCase
+  { _valueMorphismCaseOn :: GebValue,
+    _valueMorphismCaseLeft :: GebValue,
+    _valueMorphismCaseRight :: GebValue
+  }
+  deriving stock (Eq, Generic)
+
+instance NFData ValueMorphismCase
+
+data ValueMorphismBinop = ValueMorphismBinop
+  { _valueMorphismBinopOpcode :: Opcode,
+    _valueMorphismBinopLeft :: GebValue,
+    _valueMorphismBinopRight :: GebValue
+  }
+  deriving stock (Eq, Generic)
+
+instance NFData ValueMorphismBinop
+
+data ValueClosure = ValueClosure
+  { _valueClosureEnv :: Context GebValue,
+    _valueClosureLambdaBody :: Morphism
+  }
+  deriving stock (Eq, Generic)
+
+instance NFData ValueClosure
+
+instance HasAtomicity GebValue where
+  atomicity = \case
+    GebValueMorphismInteger {} -> Atom
+    GebValueMorphismLeft {} -> Aggregate appFixity
+    GebValueMorphismPair {} -> Aggregate appFixity
+    GebValueMorphismRight {} -> Aggregate appFixity
+    GebValueMorphismUnit -> Atom
+    GebValueClosure {} -> Aggregate appFixity
+
+instance Show GebValue where
```

Comment:
I think all those `Show` instances should be removed. They duplicate the pretty printing functionality in a slightly different syntax. I don't think the fact that don't print the types is enough justification to keep them.

Reply:
This was for debugging purposes. Removed.